doc/release-notes.rst

```diff
@@ -5,11 +5,276 @@
+      BlueStore for performance reasons.)  FIXME DOCS
+
+  * *Erasure coded* pools now have full support for *overwrites*,
+    allowing them to be used with RBD and CephFS.
+
+  * *ceph-mgr*:
+
+    - There is a new daemon, *ceph-mgr*, which is a required part of any
+      Ceph deployment.  Although IO can continue when *ceph-mgr* is
+      down, metrics will not refresh and some metrics-related calls
+      (e.g., ``ceph df``) may block.  We recommend deploying several instances of
+      *ceph-mgr* for reliability.  See the notes on `Upgrading`_ below.
+    - The *ceph-mgr* daemon includes a REST-based management API.  The
+      API is still experimental and somewhat limited but will form the basis
+      for API-based management of Ceph going forward.  FIXME DOCS
+
+  * The overall *scalability* of the cluster has improved. We have
+    successfully tested clusters with up to 10,000 OSDs.
+  * Each OSD can now have a *device class* associated with it (e.g., `hdd` or
+    `ssd`), allowing CRUSH rules to trivially map data to a subset of devices
+    in the system.  Manually writing CRUSH rules or manual editing of the CRUSH
+    is normally not required. FIXME DOCS
+  * You can now *optimize CRUSH weights* can now be optimized to
+    maintain a *near-perfect distribution of data* across OSDs.  FIXME DOCS
+  * There is also a new `upmap` exception mechanism that allows
+    individual PGs to be moved around to achieve a *perfect
+    distribution* (this requires luminous clients). FIXME DOCS
+  * Each OSD now adjusts its default configuration based on whether the
+    backing device is an HDD or SSD.  Manual tuning generally not required.
+  * The prototype *mclock QoS queueing algorithm* is now available.  FIXME DOCS
+  * There is now a *backoff* mechanism that prevents OSDs from being
+    overloaded by requests to objects or PGs that are not currently able to
+    process IO.
+  * There is a *simplified OSD replacement process* that is more robust.  FIXME DOCS
+  * You can query the supported features and (apparent) releases of
+    all connected daemons and clients with ``ceph features``. FIXME DOCS
+  * You can configure the oldest Ceph client version you wish to allow to
+    connect to the cluster via ``ceph osd set-require-min-compat-client`` and
+    Ceph will prevent you from enabling features that will break compatibility
+    with those clients.  FIXME DOCS
+
+- *RGW*:
+
+  * RGW introduces *server side encryption* of uploaded objects with
+    three options for the management of encryption keys: automatic
+    encryption (only recommended for test setups), customer provided
+    keys similar to Amazon SSE-C specification, and through the use of
+    an external key management service (Openstack Barbician) similar
+    to Amazon SSE-KMS specification.
+  * RGW *metadata search* with ElasticSearch now supports end user requests
+    serviced via RGW itself and now supports custom metadata fields
```

Comment:
RGW metadata search backed by ElasticSearch now supports end user requests service via RGW itself, and also supports custom metadata fields. A query language was defined and a set of RESTful APIs were created for users to be able to search objects by their metadata. New APIs that allow control of custom metadata fields were also added.